docs/source/API/core/numerics/mathematical-functions.rst

```diff
@@ -0,0 +1,215 @@
+Common math functions
+=====================
+
+.. role::cpp(code)
+    :language: cpp
+
+.. role:: strike
+    :class: strike
+
+Motivating example (borrowed from https://llvm.org/docs/CompileCudaWithLLVM.html#standard-library-support)
+
+.. code-block:: cpp
+
+    // clang is OK with everything in this function.
+    __device__ void test() {
+        std::sin(0.); // nvcc - ok
+        std::sin(0);  // nvcc - error, because no std::sin(int) override is available.
+        sin(0);       // nvcc - same as above.
+
+        sinf(0.);       // nvcc - ok
+        std::sinf(0.);  // nvcc - no such function
+    }
+
+Kokkos' goal is to provide a consistent overload set that is available on host
+and device and that follows practice from the C++ numerics library.
+
+------------
+
+.. _text: https://github.com/kokkos/kokkos/blob/develop/core/src/Kokkos_MathematicalFunctions.hpp
+
+.. |text| replace:: ``<Kokkos_MathematicalFunctions.hpp>``
+
+Defined in header |text|_ which is included from ``<Kokkos_Core.hpp>``
+
+.. _text2: https://en.cppreference.com/w/cpp/numeric/math
+
+.. |text2| replace:: standard C mathematical functions from ``<cmath>``
+
+Provides most of the |text2|_, such as ``fabs``, ``sqrt``, and ``sin``.
+
+Math functions are available in the ``Kokkos::`` namespace since version 3.7, in ``Kokkos::Experimental`` in previous versions.
+
+Below is the synopsis for ``sqrt`` as an example of unary math function.
+
+.. code-block:: cpp
+
+    namespace Kokkos {  // (since 3.7)
+        KOKKOS_FUNCTION float       sqrt ( float x );
+        KOKKOS_FUNCTION float       sqrtf( float x );
+        KOKKOS_FUNCTION double      sqrt ( double x );
+                        long double sqrt ( long double x );
+                        long double sqrtl( long double x );
+        KOKKOS_FUNCTION double      sqrt ( IntegralType x );
+    }
+
+The function is overloaded for any argument of arithmetic type. Additional functions with ``f`` and ``l`` suffixes that work on ``float`` and ``long double`` respectively are also available.  Please note, that ``long double`` overloads are not available on the device.
+
+See below the list of common mathematical functions supported. We refer the reader to cppreference.com for the synopsis of each individual function.
+
+------------
+
+:strike:`func` denotes functions that are currently not provided by Kokkos
+
+``func*`` see notes below
```

Comment:
Would it be possible to simply separate the supported vs. not supported functions with a heading?  Or even just list the supported functions, making no statement about what is not supported?  That would solve the strikethrough issue.  If Sphinx doesn't like `X` as a variable name, simply call it VariableX, or something like that.  Does this solution make sense in the context of the issues you're having?

Reply:
Hello @ajpowelsnl 

Solution 1 to separate and put a title is possible.
Solution 2 to remove the unsupported functions is possible.

It's a matter of style choice, I let you choose what you prefer.

Choice 1.1
```
# Basic operations
## Provided
abs, fabs, [and more]
## Not provided
remquo
```

Choice 1.2: Header # Not provided only here if applicable
```
# Basic operations
abs, fabs, [and more]
# Not provided
remquo
```

Choice 2 
```
# Basic operations
abs, fabs, [and more]

# Exponential functions
exp, exp2, [and more]
```

Reply:
@ajpowelsnl 
The problem is that Sphinx does not like mixing notations. A variable should either be bold, italic or strikethrough, but Sphinx doesn't like bold + italic or italic + strikethrough variables.

(My examples with x are bad, I just wanted to show the notation \`\` or ** or strikethough)

Reply:
Currently:
file .md | file .rst
:----------------: | :-------------:
![image](https://user-images.githubusercontent.com/62652182/213426381-58c44485-1971-47c5-8341-e062bb979a3c.png) | ![image](https://user-images.githubusercontent.com/62652182/213426423-be7f64a9-1484-4179-b86b-223faeb55474.png)

I'm waiting for your choice @fnrizzi & @ajpowelsnl  of style to finish!

Reply:
@antoinemeyer5 and @fnrizzi - I prefer showing what we support.  Showing what we do not support (even though it's part of the Standard) is a bit odd.  This approach takes care of the strikethrough problem, doesn't it?

Reply:
If we've implemented a subset of the standard, it's much easier on downstream developers looking for a function to see an explicit "not implemented yet", maybe with a link to where/how to request it, than to just not find anything, and have to ask/discover elsewhere what we say about missing functions.

Reply:
@PhilMiller - I can live with that -- good way to tee up work, too.

Reply:
My proposal:
live .md | rework .rst
:----------------: | :-------------:
![image](https://user-images.githubusercontent.com/62652182/213873203-ae2ea10f-c0f9-4547-8d39-4207024e17db.png) |  ![image](https://user-images.githubusercontent.com/62652182/213873206-531f0db1-be11-4e71-9788-223e008296c1.png)

I'm waiting for your opinion @fnrizzi & @ajpowelsnl & @PhilMiller to finish